ccdproc/core.py

```diff
@@ -1575,6 +1576,163 @@ def cosmicray_median(ccd, error_image=None, thresh=5, mbox=11, gbox=0,
+    and then for column sums of pixels (excluding previously flagged bad
+    pixels) from two to the number of lines in the image. The sigma of the sums
+    is scaled by the square root of the number of pixels summed so that
+    statistically low or high column regions may be detected even though
+    individual pixels may not be statistically deviant. For the purpose of this
+    task one would normally select large sigma threshold factors such as six or
+    greater to detect only true bad pixels and not the extremes of the noise
+    distribution.
+
+    As a final step each column is examined to see if there are small segments
+    of unflagged pixels between bad pixels. If the length of a segment is less
+    than that given by the ngood parameter all the pixels in the segment are
+    also marked as bad.
+
+    Returns
+    -------
+    mask : `numpy.ndarray`
+        A boolean ndarray where the bad pixels have a value of 1 (True) and
+        valid pixels 0 (False), following the numpy.ma conventions.
+    """
+    try:
+        nlines, ncols = ratio.data.shape
+    except (TypeError, ValueError):
+        # shape is not iterable or has more or less than two values
+        raise ValueError('"ratio" must be two-dimensional.')
+    except AttributeError:
+        # No data attribute or data has no shape attribute.
+        raise ValueError('"ratio" should be a "CCDData".')
+
+    def _sigma_mask(baseline, one_sigma_value, lower_sigma, upper_sigma):
+        """Helper function to mask values outside of the specified sigma range.
+        """
+        return ((baseline < -lower_sigma * one_sigma_value) |
+                (baseline > upper_sigma * one_sigma_value))
+
+    mask = ~np.isfinite(ratio.data)
+    medsub = (ratio.data -
+              ndimage.median_filter(ratio.data, size=(nlmed, ncmed)))
+
+    if byblocks:
+        nlinesblock = int(math.ceil(nlines / nlsig))
+        ncolsblock = int(math.ceil(ncols / ncsig))
+        for i in six.moves.range(nlinesblock):
+            for j in six.moves.range(ncolsblock):
+                l1 = i * nlsig
+                l2 = min((i + 1) * nlsig, nlines)
+                c1 = j * ncsig
+                c2 = min((j + 1) * ncsig, ncols)
+                block = medsub[l1:l2, c1:c2]
+                high = np.percentile(block.ravel(), 69.1)
+                low = np.percentile(block.ravel(), 30.9)
+                block_sigma = (high - low) / 2.0
+                block_mask = _sigma_mask(block, block_sigma, lsigma, hsigma)
+                mblock = np.ma.MaskedArray(block, mask=block_mask, copy=False)
+
+                if findbadcolumns:
+                    csum = np.ma.sum(mblock, axis=0)
+                    csum[csum <= 0] = 0
+                    csum_sigma = np.ma.MaskedArray(np.sqrt(c2 - c1 - csum))
+                    colmask = _sigma_mask(csum.filled(1), csum_sigma,
+                                          lsigma, hsigma)
+                    block_mask[:, :] |= colmask[None, :]
```

Comment:
Though I know it is literally the same thing, I find `np.newaxis` easier to understand when reading the code than `None`